openfl-docker/Dockerfile.base

```diff
@@ -13,28 +13,27 @@ SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 RUN --mount=type=cache,id=apt-dev,target=/var/cache/apt \
     apt-get update && \
     apt-get install -y \
+    git \
     python3-pip \
     python3.10-dev \
     ca-certificates \
     build-essential \
-    git \
     --no-install-recommends && \
+    apt-get purge -y linux-libc-dev && \
     rm -rf /var/lib/apt/lists/*
-RUN apt-get purge -y linux-libc-dev
 
 # Create an unprivileged user.
 RUN groupadd -g 1001 default && \
-    useradd -m -u 1001 -g default openfl
-USER openfl
+    useradd -m -u 1001 -g default user
+USER user
```

Comment:
What's the advantage of switching from `openfl` to `user` ?

Reply:
Eliminates confusion between the cloned repo `openfl` and `openfl` user which has a home directory named `/home/$USER`

Reply:
In this case I'd suggest using a less generic name, such as `openfl-user`. WDYT?

Reply:
or make it configurable - that way whoever run can get all under their home dir in the container as well and no permissions / user-mapping issues as well from host->container

Reply:
defaulted to generic name like @teoparvanov suggested

Reply:
Docker allows specifying `--user` with `docker run`. I think that covers?

Reply:
Yep, sounds good.

Reply:
Haven't used `--user` before personally, hence the comment; I hope having `WORKDIR /home/user` in Dockerfile would not affect the container, permissions or otherwise, if any other user is specified while running.

Reply:
To an extent this is inevitable. AFAIK, a `USER` defined in the container should match the `--user` that gets passed at runtime (+ same uid/gid) to avoid permission issues.

Since any guess on our part would be just as good (or bad), we should leave it up to the users to edit their Dockerfile and match their deployments.